sycl/include/sycl/ext/oneapi/work_group_scratch_memory.hpp

```diff
@@ -0,0 +1,70 @@
+//==--- work_group_scratch_memory.hpp - SYCL group local memory extension --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include <sycl/detail/defines_elementary.hpp> // for __SYCL_ALWAYS_INLINE
+#include <sycl/detail/sycl_builtins.hpp>      // for __sycl_allocateLocalMemory
+#include <sycl/exception.hpp>                 // for exception
+#include <sycl/ext/oneapi/properties/properties.hpp> // for properties
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi {
+namespace experimental {
+
+__SYCL_ALWAYS_INLINE
+inline void *get_work_group_scratch_memory() {
+#ifdef __SYCL_DEVICE_ONLY__
+  return __sycl_dynamicLocalMemoryPlaceholder();
+#else
+  throw sycl::exception(
+      sycl::errc::feature_not_supported,
+      "sycl_ext_oneapi_work_scratch_memory extension is not supported on host");
+#endif
+}
+
+// Property
+struct work_group_scratch_size
+    : ::sycl::ext::oneapi::experimental::detail::run_time_property_key<
+          work_group_scratch_size,
+          ::sycl::ext::oneapi::experimental::detail::WorkGroupStaticMem> {
```

Comment:
```suggestion
          ::sycl::ext::oneapi::experimental::detail::PropKind::WorkGroupStaticMem> {
```

I'm surprised it's not `enum class`, but it's probably because old properties design benefited from implicit casts to int...

Also, why mismatch in names between the key and the enum kind?